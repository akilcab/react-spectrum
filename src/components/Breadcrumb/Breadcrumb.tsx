import React from "react";
import {
  Breadcrumbs,
  createStyles,
  Grid,
  makeStyles,
  Paper,
  Typography,
} from "@material-ui/core";

export interface BreadcrumbProps {
  title: string;
  secondLabel: string;
}

const useStyles = makeStyles(() =>
  createStyles({
    breadcrumbPaper: {
      background: "transparent",
    },
  })
);

export const Breadcrumb: React.FC<BreadcrumbProps> = (props) => {
  const { title, secondLabel } = props;
  const classes = useStyles();

  return (
    <Grid item xs={12}>
      <Paper className={classes.breadcrumbPaper} elevation={0}>
        <Typography variant="h3">{title}</Typography>
        <Breadcrumbs aria-label="breadcrumb">
          <Typography color="inherit">{title}</Typography>
          <Typography color="secondary">{secondLabel}</Typography>
        </Breadcrumbs>
      </Paper>
    </Grid>
  );
};
