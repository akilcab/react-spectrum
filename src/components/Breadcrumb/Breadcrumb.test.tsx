import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Breadcrumb } from "./Breadcrumb";

describe("<Breadcrumb />", () => {
  test("rend correctement", () => {
    const component = renderer
      .create(
        <Breadcrumb
          title="Comptes connectés"
          secondLabel="Liste des comptes connectés"
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test("should have content texts in the component", () => {
    const component = render(
      <Breadcrumb
        title="Comptes connectés"
        secondLabel=" Liste des comptes connectés"
      />
    );
    expect(component.container).toHaveTextContent("Comptes connectés");
    expect(component.container).not.toHaveTextContent("Comptes non connectés");
  });

  test("try to display the profile without its require props", () => {
    expect(Breadcrumb).toThrowError();
  });
});
