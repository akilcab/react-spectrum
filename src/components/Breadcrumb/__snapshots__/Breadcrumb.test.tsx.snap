// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> rend correctement 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
>
  <div
    className="MuiPaper-root makeStyles-breadcrumbPaper-1 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3"
    >
      Comptes connectés
    </h3>
    <nav
      aria-label="breadcrumb"
      className="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <ol
        className="MuiBreadcrumbs-ol"
      >
        <li
          className="MuiBreadcrumbs-li"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
          >
            Comptes connectés
          </p>
        </li>
        <li
          aria-hidden={true}
          className="MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          className="MuiBreadcrumbs-li"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorSecondary"
          >
            Liste des comptes connectés
          </p>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;
